crates/runtime/src/dialogue.rs

```diff
@@ -325,71 +309,90 @@ impl Dialogue {
-        if let Some(program) = self.program() {
-            program.nodes.contains_key(node_name)
-        } else {
-            self.log_error_message
-                .call("Tried to call NodeExists, but no program has been loaded".to_owned());
-            false
-        }
+        self.handler_safe_dialogue.node_exists(node_name)
     }
 
-    pub fn analyse(&mut self) {
-        // ## Implementation notes
-        // It would be more ergonomic to not expose this and call it automatically.
-        // We should probs remove this from the API.
-        // Call it when running the first `continue_` after adding a program or something.
-        todo!()
+    /// Replaces all substitution markers in a text with the given
+    /// substitution list.
+    ///
+    /// This method replaces substitution markers - for example, `{0}`
+    /// - with the corresponding entry in `substitutions`.
+    /// If `test` contains a substitution marker whose
+    /// index is not present in `substitutions`, it is
+    /// ignored.
+    pub fn expand_substitutions<'a>(
+        text: &str,
+        substitutions: impl IntoIterator<Item = &'a str>,
+    ) -> String {
+        HandlerSafeDialogue::expand_substitutions(text, substitutions)
     }
 
-    pub fn parse_markup(&mut self) {
-        // ## Implementation notes
-        // It would be more ergonomic to not expose this and call it automatically.
-        // We should probs remove this from the API.
-        // Pass the MarkupResult directly into the LineHandler
-        todo!()
+    /// Gets the name of the node that this Dialogue is currently executing.
+    ///
+    /// If [`Dialogue::continue_`] has never been called, this value
+    /// will be [`None`].
+    pub fn current_node(&self) -> Option<String> {
+        self.handler_safe_dialogue.current_node()
     }
 
-    fn get_node_logging_errors(&self, node_name: &str) -> Option<&Node> {
-        if let Some(program) = self.program() {
-            if program.nodes.is_empty() {
-                self.log_error_message
-                    .call("No nodes are loaded".to_owned());
-                None
-            } else if let Some(node) = program.nodes.get(node_name) {
-                Some(node)
-            } else {
-                self.log_error_message
-                    .call(format!("No node named {node_name}"));
-                None
-            }
-        } else {
-            self.log_error_message
-                .call("No program is loaded".to_owned());
-            None
-        }
+    /// The [`Dialogue`]'s locale, as an IETF BCP 47 code.
+    ///
+    /// This code is used to determine how the `plural` and `ordinal`
+    /// markers determine the plural class of numbers.
+    ///
+    /// For example, the code "en-US" represents the English language as
+    /// used in the United States.
+    pub fn language_code(&self) -> Option<String> {
```

Comment:
You know what represents Zürich German?

> gsw-u-sd-chzh

yes. That's correct.

Reply:
(cuz it doesn't have a [ISO 639-3](https://en.wikipedia.org/wiki/ISO_639-3))

Reply:
(Ignore this comment I just want to amuse you ;))

Reply:
phahahaha nice!